lightning/src/ln/channelmanager.rs

```diff
@@ -13048,6 +13057,17 @@ where
 			Some(hrn) => builder.sourced_from_human_readable_name(hrn),
 		};
 
+		let contacts = match contacts {
+			None => ContactSecrets::new(self.entropy_source.get_secure_random_bytes()),
+			Some(c) => c,
+		};
+		let builder = builder.contact_secrets(contacts.clone());
+		// Create a minimal offer for BLIP-42 contact exchange (just node_id, no description/paths)
+		// TODO: Create a better minimal offer with a single blinded path hop for privacy,
+		// while keeping the size small enough to fit in the onion packet.
```

Comment:
Luckily the onion packet is 64K so we're not too worried about the size. Indeed, as you note in your questions, we should probably try to figure out if the dev probably wants us to use an async offer (based on async availability/config) and use one of those if we can.